pkg/systemtemplate/import.go

```diff
@@ -70,30 +72,39 @@ func toFeatureString(features map[string]bool) string {
 }
 
 func SystemTemplate(resp io.Writer, agentImage, authImage, namespace, token, url string, isWindowsCluster bool,
-	cluster *apimgmtv3.Cluster, features map[string]bool, taints []corev1.Taint, secretLister v1.SecretLister) error {
-	var tolerations, agentEnvVars string
-	d := md5.Sum([]byte(url + token + namespace))
-	tokenKey := hex.EncodeToString(d[:])[:7]
+	cluster *apimgmtv3.Cluster, features map[string]bool, tolerations []corev1.Toleration, affinity *corev1.Affinity,
+	resourceRequirements *corev1.ResourceRequirements, secretLister v1.SecretLister) error {
+	var agentEnvVars string
+	var agentTolerations, agentAffinity, agentResourceRequirements string
+	envVars := settings.DefaultAgentSettingsAsEnvVars()
+	if cluster != nil {
+		envVars = append(envVars, cluster.Spec.AgentEnvVars...)
+	}
+	agentEnvVars = templates.ToYAML(envVars)
 
 	if authImage == "fixed" {
 		authImage = settings.AuthImage.Get()
 	}
 
-	registryURL, registryConfig, err := util.GeneratePrivateRegistryEncodedDockerConfig(cluster, secretLister)
-	if err != nil {
-		return err
+	d := md5.Sum([]byte(url + token + namespace))
```

Comment:
Not sure what's the rationale behind shuffling around lines here - makes reviewing harder to see what actually got changed.